File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# a0 = 1
# a1 = a
# an = a * ... * a (n-czynników) dla n należących do N+ - {0,1}

def potega_it(podst, wykladnik):
    """Funkcja oblicza iteracyjnie potęgę l. naturalnej"""
    wynik = 1
    for i in range(wykladnik):
        wynik *= podst
    return wynik


def main(args):
    # pobierz od użytkownika podstawe i wykładnik i przypisz do odpowiednich zmiennych
    # wywołaj funkcje potega_it()
    podst = int(input("Podaj podstawę: "))
    wykladnik = int(input("Podaj wykładnik: "))
    assert type(podst) == int
    assert type(wykladnik) == int

    assert potega_it(100, 0) == 1
    assert potega_it(100, 1) == 100
    assert potega_it(2, 3) == 8

    print("Potęga = ", potega_it(podst, wykladnik))
    #wynik = potega_it(podst, wykladnik)
    #print("Potęga = ", wynik)

    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/potega.cpp
/*
 * potega.cpp
 * 
 * a0 = 1
 * a1 = a
 * an = a * ... * a (n-czynników) dla n zaw. N+ - {1}
 */


#include <iostream>

using namespace std;


int potega_rek(int x, int n)
{    
    if (x == 0)
        return 0;
    else if (n == 0)
        return 1;
    return potega_rek(x, n - 1) * x;
}

int main(int argc, char **argv)
{
    int n;
    float x;
    cout << "Podaj podstawe potęgi: " << endl;
    cin >> x;
    cout << "Podaj wykładnik potęgi: " << endl;
    cin >> n;
    cout << "Wynik: " << potega_rek(x, n) <<endl;
	return 0;
}
